src/coreclr/jit/gentree.cpp

```diff
@@ -10213,6 +10213,117 @@ void Compiler::gtUpdateNodeSideEffects(GenTree* tree)
     });
 }
 
+//------------------------------------------------------------------------
+// gtSplitEffectsForSubTree: Split any effect for a subtree to make sure the subtree
+//                           is the first executed effect in the statement.
+//
+// Arguments:
+//    stmt - The statement to spilt the side effects for
+//    subTree - The subtree to be the first executed effect
+//    early - Whether the run is early, in which case we need to check lvHasLdAddrOp as
+//            we don't have valid GTF_GLOB_REF yet
+//
+// Returns:
+//    The first new statement that is inserted for the split effects.
+//
+// Notes:
+//    This method requires compCurBB to be set to the block that contains the statement.
+//
+Statement* Compiler::gtSplitEffectsForSubTree(Statement* stmt, GenTree* subTree, bool early)
```

Comment:
`gtSplitTree` exists already

Reply:
Oops didn't notice that before. Does it always split the tree even if the tree is the first executed effect?

Reply:
Okay seems that it works in a different way. Switched to use `gtSplitTree` instead.